src/Item/BaseItem.php

```diff
@@ -25,4 +25,21 @@ public static function getEndpoint()
     {
         return '/'.snake_case(str_plural(substr(get_called_class(), strrpos(get_called_class(), '\\') + 1)));
     }
+
+    public function toJsonApiArray(): array
+    {
+        $attributes = parent::toJsonApiArray();
+        $whitelist = $this->getWhiteListAttributesKeys();
+        if (is_array($whitelist) && isset($attributes['attributes'])) {
+            // keep only whitelisted attributes
+            $attributes['attributes'] = array_intersect_key($attributes['attributes'], array_flip($whitelist));
+        }
```

Comment:
a bit confusing as for me. will be better to track creatableAttributes & updatableAttributes (isNew()=>false) instead of getWhiteListAttributesKeys (has no difference between creation & updating). also getWhiteListAttributesKeys can return null (prevent whitelisting) or array